pyccel/codegen/printing/ccode.py

```diff
@@ -1600,13 +1600,13 @@ def _print_Return(self, expr):
                 # make sure that stmt contains one assign node.
                 last_assign = last_assign[-1]
                 variables = last_assign.rhs.get_attribute_nodes(Variable)
-                unneeded_var = not any(b in vars_in_deallocate_nodes for b in variables)
+                unneeded_var = not any(b in vars_in_deallocate_nodes or b.is_ndarray for b in variables)
                 if unneeded_var:
                     code = ''.join(self._print(a) for a in expr.stmt.body if a is not last_assign)
                     return code + 'return {};\n'.format(self._print(last_assign.rhs))
                 else:
-                    code = ''+self._print(expr.stmt)
                     last_assign.lhs.is_temp = False
+                    code = ''+self._print(expr.stmt)
```

Comment:
I'm not sure why you moved this line, but as you did you must have looked at it. Do you know why it is `code = ''+self._print(expr.stmt)` and not simply `code = self._print(expr.stmt)`?

Reply:
I swapped the line order because the last assign should become a non temporary variable for it to be declared, if a loop unraveling happens during the printing the last assign lhs will become an indexed element that doesn't have the attribute is_temp, that's why i swapped order.
And i agree it should be `code = self._print(expr.stmt)` i'll fix that.

Reply:
Fixed in [52d0f45](https://github.com/pyccel/pyccel/pull/1231/commits/52d0f4520b9dfafb8462427d007880650990f96a).